### ISSUE TITLE
Type inference failing in recent versions

### ISSUE THREAD

**Issue by kyle-query** [microsoft/pyright#5636]
**Describe the bug**

I've encountered a couple cases where a compound expression results in a type error, but splitting it into subexpressions stored in temporary variables does not. One such example:

```python
from random import choice
from typing import Literal, get_args

A = Literal["a"]

def test0() -> A:
    return choice(get_args(A)) # type error

def test1() -> A:
    a = choice(get_args(A))
    return a # OK
```

The error is:

```
Diagnostics:
1. Argument of type "tuple[Any, ...]" cannot be assigned to parameter "seq" of type "SupportsLenAndGetItem[_T@choice]" in function "choice"
     "tuple[Any, ...]" is incompatible with protocol "SupportsLenAndGetItem[A]"
       Type parameter "_T_co@SupportsLenAndGetItem" is covariant, but "A" is not a subtype of "A" [reportGeneralTypeIssues]
```

Interestingly this does not generate a type error:

```python
from random import choice
from typing import Literal, get_args

B = Literal["b0", "b1"]

def test2() -> B:
    return choice(get_args(B))
```

**VS Code extension or command-line**

This issue was introduced in 1.1.314 and exists in the latest release, 1.1.320

**Comment by erictraut** [comment]
Thanks for the bug report. This will be addressed in the next release.

**Comment by erictraut** [comment]
This is addressed in pyright 1.1.321, which I just published. It will also be included in a future release of pylance.
